api/src/controllers/correction-request.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * /correction-request
+ *
+ * ROUTES
+ *   POST   /:youngId -> create new requests
+ */
+
+const express = require("express");
+const router = express.Router({ mergeParams: true });
+const Joi = require("joi");
+
+const YoungModel = require("../models/young");
+const { capture } = require("../sentry");
+const { serializeYoung } = require("../utils/serializer");
+
+const { ERRORS } = require("../utils");
+
+const { validateFirstName } = require("../utils/validator");
+const passport = require("passport");
+
+router.post("/:youngId", passport.authenticate("referent", { session: false, failWithError: true }), async (req, res) => {
+  try {
+    const { error: error_id, value: youngId } = Joi.string().required().validate(req.params.youngId, { stripUnknown: true });
+    if (error_id) return res.status(400).send({ ok: false, code: ERRORS.INVALID_PARAMS });
+
+    const { error, value: newRequests } = Joi.array()
+      .items(
+        Joi.object({
+          cohort: validateFirstName().trim().required(),
+          field: Joi.string().required(),
+          reason: Joi.string(),
+          message: Joi.string(),
+          status: Joi.string().valid("PENDING", "SENT", "REMINDED", "CORRECTED", "CANCELED").required(),
+        }),
+      )
+      .validate(req.body, { stripUnknown: true });
+
+    if (error) {
+      return res.status(400).send({ ok: false, code: ERRORS.INVALID_BODY });
+    }
+
+    const young = await YoungModel.findById(youngId);
+    if (!young) return res.status(404).send({ ok: false, code: ERRORS.NOT_FOUND });
+
+    const sentAt = Date.now();
+    for (const request of newRequests) {
+      request.moderatorId = req.user._id;
+      request.sentAt = sentAt;
+      request.status = "SENT";
```

Comment:
On réécrit le status ?
Pas mieux de mettre "SENT" if status is null or XXX ?

Reply:
A ce moment là, on traite les nouvelles demandes donc elles ne peuvent que être SENT. Il y aura certainement un petit check supplémentaire lorsque je vais traiter les deuxièmes demandes.